pages/add/add.jsx

```diff
@@ -1,12 +1,201 @@
 import React from 'react';
+import { browserHistory } from 'react-router';
+import HintMessage from '../../components/hint-message/hint-message.jsx';
+import Service from '../../js/service.js';
+import UserData from '../../js/user-data.js';
+import env from "../../config/env.generated.json";
+
+const CreatorInputField = () => {
+  return (
+    <input name="creators" type="text" placeholder="Name" width="150" className="form-control" />
+  );
+};
 
 export default React.createClass({
+  getInitialState() {
+    return {
+      numCreatorFields: 1,
+      loggedInStatusChecked: false
+    };
+  },
+  componentWillMount() {
+    console.log(`UserData.userLoggedInStatus.get() = `, UserData.userLoggedInStatus.get());
+  },
+  componentDidMount() {
+    let loggedInParam = this.props.router.location.query.loggedin;
+
+    if ( !loggedInParam || loggedInParam.toString().toLowerCase() === `true` ) {
+      // if there's no "loggedin" param presented in the URL OR if "loggedin" param is true
+      // we want to verify if user is really logged in by making a /userstatus call to Pulse API
+      this.verifyLoggedInStatus();
```

Comment:
This leads to the wrong codepath. Right now with this code in place, the form page will do a user login check, and when the user isn't logged in says "Sign in failed" which is wrong: no sign in even occurred. Instead when there is no `loggedin` query argyment, this code should says "Please log in to post a new entry" or some such, with the same link we present on the `favs` page.